### ISSUE TITLE
[EPIC] Flipper audit -Identify and remove unnecessary Flippers 

### ISSUE THREAD

**Issue by dsinla** [department-of-veterans-affairs/va.gov-cms#10656]
## Description

All the detail in previous ticket: Follow up from https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10350

List of flippers and owners: https://docs.google.com/spreadsheets/d/1yQb5oDshNj7y0nvtF3PaZ9OLhtTuZ2aO8t4_n51YBWU/edit#gid=237722104

Tracking here for at-a-glance status

| **Number** | **Flipper**                                                       | **Status**   | **Decision** | **Ticket for removal** |
|------------|-------------------------------------------------------------------|--------------|--------------|------------------------|
| 1          | covid_vaccine_scheduling_frontend                                 | INVESTIGATED | REMOVE       |                        |
| 2          | facilities_locator_mobile_covid_online_scheduling                 | TBD          |              |             #11588            |
| 3          | facilities_ppms_suppress_all                                      |              | KEEP         |    #11589    |  
| 4          | facilities_ppms_suppress_community_care                           |              | KEEP         |        #11590              |
| 5          | facilities_ppms_suppress_pharmacies                               |              | KEEP         |      #11591                  |
| 6          | facility_locator_covid_vaccine_walk_in_availability_text_frontend |   TBD           |              |                        |
| 7          | facility_locator_lat_long_only                                    | INVESTIGATED | REMOVE       |                        |
| 8          | facility_locator_predictive_location_search                       | INVESTIGATED | REMOVE       |                        |
| 9          | facility_locator_show_operational_hours_special_instructions      | INVESTIGATED | REMOVE       |                        |
| 10         | facility_locator_lighthouse_covid_vaccine_query                   | TBD          |              |     #11592                   |
| 11         | facility_locator_ppms_forced_unique_id                            | TBD          |              |     #11593                   |
| 12         | facility_locator_ppms_legacy_urgent_care_to_pos_locator           | TBD          |              |          #11594             |
| 13         | facility_locator_pull_operating_status_from_lighthouse            | TBD          |              |       #11595                 |
| 14         | facility_locator_rails_engine                                     | TBD          |              |            #11596            |  
| 15         | facility_locator_restore_community_care_pagination                | TBD          |              |      #11597                  |
| 16         | facility_locator_show_community_cares                             | INVESTIGATED | REMOVAL IN PROGRESS             |    #11598 #14112                   |
| 17         | facility_locator_show_health_connect_number                       | TBD          |              |      #11599                  |
| 18         | show_expandable_vamc_alert                                        | TBD          |              |      #11600                  |
| 19         | caregiver_use_facilities_API                                      | INVESTIGATED | KEEP         |       #11601                 |

## Acceptance Criteria
- [ ] Flippers owned by Facilities have been identified and evaluated
- [ ] Flippers that are no longer required are removed

**Comment by dsinla** [comment]
Message sent to @jilladams to get the master list

**Comment by jilladams** [comment]
The doc we created when went through the lists together is an XLS in sharepoint now, as that's the only way to share spreadsheets with Dave: https://dvagov-my.sharepoint.com/:x:/g/personal/jill_adams1_va_gov/EWJYp5QIYQ5Bng7ynlMX9hIBxXjsVBNANaugiektKZ1fcQ

https://dsva.slack.com/archives/C52CL1PKQ/p1663958562870949

**Comment by jilladams** [comment]
Apologies for the confusion today - Google sheet link was added to our ticket, not yours. 
https://docs.google.com/spreadsheets/d/1yQb5oDshNj7y0nvtF3PaZ9OLhtTuZ2aO8t4_n51YBWU/edit#gid=237722104&fvid=1552255965

Updated the related slack thread for the histories as well.

**Comment by dsinla** [comment]
thanks @jilladams 
@maxx1128  please take a look - looks like ~20 items.

**Comment by maxx1128** [comment]
Thanks @jilladams! Going to tackle these

* caregiver_use_facilities_API
* covid_vaccine_scheduling_frontend
* facilities_locator_mobile_covid_online_scheduling
* facilities_ppms_suppress_all
* facilities_ppms_suppress_community_care
* facilities_ppms_suppress_pharmacies
* facility_locator_covid_vaccine_walk_in_availability_text_frontend
* facility_locator_lat_long_only
* facility_locator_predictive_location_search
* facility_locator_show_operational_hours_special_instructions

**Comment by maxx1128** [comment]
> Thanks @jilladams! Going to tackle these
> 
> * caregiver_use_facilities_API
> * covid_vaccine_scheduling_frontend
> * facilities_locator_mobile_covid_online_scheduling
> * facilities_ppms_suppress_all
> * facilities_ppms_suppress_community_care
> * facilities_ppms_suppress_pharmacies
> * facility_locator_covid_vaccine_walk_in_availability_text_frontend
> * facility_locator_lat_long_only
> * facility_locator_predictive_location_search
> * facility_locator_show_operational_hours_special_instructions

Sorry for the delay, I looked through the Facilities flippers for which ones can be safely removed. These aren't being used on the front end so they can safely be removed:

* `facility_locator_covid_vaccine_walk_in_availability_text_frontend`
* `facilities_locator_mobile_covid_online_scheduling`

**Comment by dsinla** [comment]
@maxx1128 @jilladams  Does Facilities perform the removal? - or do we request some other team to remove?

**Comment by jilladams** [comment]
Facilities owns the code / downstream impact for those flippers, so can remove any that you're confident are ready to be removed.

**Comment by dsinla** [comment]
@maxx1128 do you feel confident to remove the flippers? Do you want to run it by @swirtSJW  for a quick sanity check first?

**Comment by dsinla** [comment]
These must stay:

- facilities_ppms_suppress_all
- facilities_ppms_suppress_community_care
- facilities_ppms_suppress_pharmacies

**Comment by maxx1128** [comment]
@dsinla I just finished looking more into the two feature flags that needed more investigation and I could do right away, `covid_vaccine_scheduling_frontend` and `facility_locator_show_operational_hours_special_instructions`.

The functionality connected to both flags, COVID scheduling, and operational hour special instructions, have extra checks that ensure the info/functionality is first available before showing them. So if they were removed, the code would always let the info/functionality be shown if it was set - if they're ready to be turned on for everyone, then removing their code is a relatively easy change. I don't know if doing this would require any extra approval or if it's already been decided.

Long story short - the code around these flags should prevent any errors if they're removed from the code and will be easy to remove. The big question is just if everyone should be allowed to see it yet.

**Comment by dsinla** [comment]
@maxx1128  Thanks for investigating. Regarding:
_If they're ready to be turned on for everyone, then removing their code is a relatively easy change._
I'm not sure who _everyone_ is in this scenario. Let's discuss in 16th minute with the rest of the dev team tomorrow

**Comment by mmiddaugh** [comment]
@dsinla

re: facility_locator_lat_long_only- I believe this was turned off after an unexpected consequence was identified
- Reference [Front end to send lat/long for queries (1a)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/31527#top)

**Comment by maxx1128** [comment]
Initial investigative report on three of the feature flags cc: @dsinla 

### `covid_vaccine_scheduling_frontend`

**Status:** Off
**Issue:** [#22760](https://github.com/department-of-veterans-affairs/va.gov-team/issues/22760)
**Ticket Author:** Michelle Middaugh
**Code Author:** Michael Pelz-Sherman
**Purpose:** Only show COVID vaccine scheduling in facilities that allow it as a service through VAOS. 

### `facility_locator_show_operational_hours_special_instructions`

**Status:** Off
**Issue:** [#19615](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19615)
**Ticket Author:** Michelle Middaugh
**Code Author:** Gilbran
**Purpose:** Special language for facility hours can be added in Lighthouse. There are four potential special instructions to add, and multiple can be added.

### `caregiver_use_facilities_API`

**Status:** Off
**Issue:** [#26246](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26246)
**Ticket Author:** Chris Dyer
**Code Author:** Matt Long
**Purpose:** Let users select the most relevant VA Medical Facility with a Caregiver Support Coordinator so they can participate in the PCAFC (couldn't find this acronym).

**Comment by dsinla** [comment]
Thanks @maxx1128 

In comments above @mmiddaugh indicates that` facility_locator_lat_long_only `and `facility_locator_predictive_location_search `were never implemented or turned on, so it seems conclusive that you can remove them. 

Note: PCAFC =  Program of Comprehensive Assistance for Family Caregivers from https://www.caregiver.va.gov/support/support_benefits.asp

**Comment by jilladams** [comment]
Discovered today that I had locked [the google sheet ](https://docs.google.com/spreadsheets/d/1yQb5oDshNj7y0nvtF3PaZ9OLhtTuZ2aO8t4_n51YBWU/edit#gid=237722104)so no one coudl edit it. That's fixed now, and I've ported notes from comments here into the sheet as well so folks can see the rundown.  You all should be able to edit it moving forward, if necessary / helpful.

**Comment by dawnpruitt** [comment]
@mmiddaugh Hello! Would you please reply with your approval to remove the following? Details for each are above in the ticket. Thank you!
caregiver_use_facilities_API
facility_locator_show_operational_hours_special_instructions
covid_vaccine_scheduling_frontend

**Comment by maxx1128** [comment]
Updating this comment as I get some branches ready to open up once these flippers get approval to remove, either in this ticket or a future one.

`caregiver_use_facilities_API`: `remove-caregiver-use-facilities-api-flipper`

**Comment by dawnpruitt** [comment]
**Please do not remove caregiver_use_facilities_API**
[I reached out to the 1010-health-apps slack channel to request review of this flag:](https://dsva.slack.com/archives/CMJ2V70UV/p1667403093401779)

Matt Long
Hi, Dawn! I’m the one that implemented that toggle for the CG app. There is also a subsequent one for the implementation on the HCA app as well.

Dawn Pruitt
Hi Matt! Nice to meet you! We are showing that it's off but I recognize that doesn't mean it's not needed. Would you please let me know if it can be removed or if you'd like for us to leave it be?

Matt Long
Nice to meet you as well! We’d like to leave those in place. We created an MVP to ensure we could get the data we needed from the API and are currently in the planning stages to do a full-featured implementation within both forms! :upside_down_face: (edited) 

Dawn Pruitt
Thank you so much for your quick response! I will make a note in our ticket to leave it in place and let our engineers know.

Matt Long
Sounds great! Thank you for reaching out!

**Comment by dsinla** [comment]
@maxx1128  @dawnpruitt Trying to think of the best way to track this. Jill has a google sheet linked to from this ticket. 

It looks like we know what to do with 8 of these, but still have 10 that have yet to be investigated TBD.  It looks like these investigations take time - while we wait for approval, and we may have this ticket for at least one more sprint if not 2. I also wanted to be able to look at this ticket to understand where we are with each ticket.  I added a table in the Description above.  Would it be reasonable to update this ticket as we get new info for the remaining flippers?

**Comment by dawnpruitt** [comment]
@maxx1128 is there any change to status on this one? Thanks!

**Comment by dsinla** [comment]
@dawnpruitt 
These 3 are ready from Max's [comment above](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10656#issuecomment-1298909446) 

> Updating this comment as I get some branches ready to open up once these flippers get approval to remove, either in this ticket or a future one.
> 
> caregiver_use_facilities_API: remove-caregiver-use-facilities-api-flipper
> facility_locator_show_operational_hours_special_instructions: remove-facility-hours-special-instructions
> covid_vaccine_scheduling_frontend: remove-covid-scheduling-frontend-flipper

Did you send messages to owners? ..are waiting for some approval?

**Comment by dawnpruitt** [comment]
Please do not remove the  caregiver_use_facilities_API -as noted above, the health apps team does not want that to be removed.

Regarding the other two, I was under the impression they were owned by the previous Facilities team, but we are now that team and can make the decision. 

I have not taken any action on any of the other flippers.

**Comment by dsinla** [comment]
Quick chat with @mmiddaugh she will take a look at covid_vaccine_scheduling_frontend

**Comment by mmiddaugh** [comment]
It looks like [Max has investigated the flag](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10656#issuecomment-1290901435) for covid_vaccine_scheduling_frontend. This is another situation where the functionality was built but only briefly implemented. In this case, I believe this was waiting for backend integration with VAOS. Meanwhile, the need for appointments was quickly replaced by access to vaccines on a walk-in basis so the functionality was retained but behind a flipper.

**Comment by dsinla** [comment]
Thanks @mmiddaugh! I think i was confusing some of the very-similar names, but would like you to weigh in on one more: facility_locator_show_operational_hours_special_instructions

@maxx1128[ has investigated](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10656#issuecomment-1290901435), identified you as the owner (at one point). As with covid_vaccine_scheduling_frontend we want someone with more background to weigh in before we remove it.

**Comment by davidmpickett** [comment]
@jilladams I'm confused why this is in the Current Sprint, Not started column for Facilities. Seems like a mistake

**Comment by randimays** [comment]
All sub-issues are complete; closing epic as complete!
